webpack/assets/javascripts/react_app/components/ForemanModal/index.js

```diff
@@ -24,7 +26,7 @@ const ConnectedForemanModal = props => {
 
   useEffect(() => {
     if (modalAlreadyExists) return; // don't add modal if it already exists
-    dispatch(addModal({ id, open: false }));
+    dispatch(addModal({ id, open: false, submitting: false }));
```

Comment:
would it be a better performance to declare this function outside the useEffect, like the `onClose` function?

Reply:
@laviro I don't think this is necessary, see [this part](https://reactjs.org/docs/hooks-faq.html#are-hooks-slow-because-of-creating-functions-in-render) of the docs where React weighs in on it.  They do provide some other hooks to increase performance, but I think any performance optimization effort isn't really worth it unless performance is measured first.